### PR TITLE
black: update resources

### LINKED ISSUE #56328: Missing dependencies of black
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [ ] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

### Output of `brew gist-logs black`

```
Error: No logs.
```

### Output of `brew config`

```
HOMEBREW_VERSION: 2.4.0-41-g0125c4c
ORIGIN: https://github.com/Homebrew/brew
HEAD: 0125c4cd56b1e9a238206090dbf3f6a24a463c7d
Last commit: 22 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 7a12c4e47ad249f720c32caa3f64060f9dff619e
Core tap last commit: 4 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_MAKE_JOBS: 8
CPU: octa-core 64-bit kabylake
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
Clang: 11.0 build 1103
Git: 2.24.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 10.15.5-x86_64
CLT: 11.5.0.0.1.1588476445
Xcode: N/A
```

## What you were trying to do (and why)

```bash
brew install black
blackd
```

to run [blackd](https://black.readthedocs.io/en/stable/blackd.html)

## What happened (include command output)

1. After brew install

```
Traceback (most recent call last):
  File "/usr/local/bin/blackd", line 5, in <module>
    from blackd import patched_main
  File "/usr/local/Cellar/black/19.10b0_2/libexec/lib/python3.8/site-packages/blackd.py", line 9, in <module>
    from aiohttp import web
ModuleNotFoundError: No module named 'aiohttp'
```

2. After `/usr/local/Cellar/black/19.10b0_2/libexec/bin/python3.8 -m pip install aiohttp`

```
Traceback (most recent call last):
  File "/usr/local/bin/blackd", line 5, in <module>
    from blackd import patched_main
  File "/usr/local/Cellar/black/19.10b0_2/libexec/lib/python3.8/site-packages/blackd.py", line 10, in <module>
    import aiohttp_cors
ModuleNotFoundError: No module named 'aiohttp_cors'
```

3. After `/usr/local/Cellar/black/19.10b0_2/libexec/bin/python3.8 -m pip install aiohttp_cors`

```
blackd version 19.10b0 listening on localhost port 45484
INFO:aiohttp.access:127.0.0.1 [15/Jun/2020:05:42:30 +0000] "POST / HTTP/1.1" 200 13515 "-" "Java/11.0.7"
```

## What you expected to happen

```
blackd version 19.10b0 listening on localhost port 45484
INFO:aiohttp.access:127.0.0.1 [15/Jun/2020:05:42:30 +0000] "POST / HTTP/1.1" 200 13515 "-" "Java/11.0.7"
```

without installing extra dependencies manually.

## Step-by-step reproduction instructions (by running `brew install` commands)

```bash
brew install black
blackd
```

---

I'm a newbie in MacOS and Homebrew. Could you please add those dependencies to the formula?

### PR DESCRIPTION
**Pull request by linzichao**
Issue #56328
Add resources for black.

- [*] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [*] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [*] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [*] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [*] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Review by iMichka — DISMISSED** [review]
Hi. Can you please update the `revision` line from 2 to 3, so that the new resources get shipped to everyone on the next upgrade?

**Comment by linzichao** [comment]
@iMichka modify as you advice

**Comment by dawidd6** [comment]
I would say we should be running `blackd` in test block too. So we can discover breakages like this easier.

**Comment by linzichao** [comment]
@dawidd6 
I add the blackd test in the commit dfa04b1051842b6a50d960b69e29c019237a7124.

**Comment by dawidd6** [comment]
Please squash your commits and it will be good to go.

**Comment by dtrodrigues** [comment]
While these dependencies are being added, would it make sense to update the other python dependencies?
appdirs 1.4.3 -> 1.4.4
click 7.0 -> 7.1.2
pathspec 0.6.0 -> 0.8.0
regex 2019.11.1 -> 2020.6.8
toml 0.10.0 -> 0.10.1
typed_ast 1.4.0 -> 1.4.1

**Comment by dtrodrigues** [comment]
Should the test be using the `free_port` helper instead of hard-coding port 9090 in case something else is using port 9090?

**Comment by linzichao** [comment]
According to @dtrodrigues advice, I update the packages version and use ```free_port```.
Everything run as usual.

File: Formula/black.rb [hunk 1]
```diff
@@ -63,5 +98,9 @@ def install
     EOS
     system bin/"black", "black_test.py"
     assert_equal "print(\"It works!\")\n", (testpath/"black_test.py").read
+    port = "#{free_port}"
```
Comment (SMillerDev) [inline Formula/black.rb:101]:
```suggestion
    port = free_port
```

Comment (linzichao) [reply Formula/black.rb]: thanks for reminder

**Review by dawidd6 — APPROVED** [review]

**Review by issyl0 — APPROVED** [review]
Thanks for your first contribution to Homebrew, @linzichao!

**Pull request closed without merging**